### PR TITLE
refactor: 💡 Improve hotkeyPreferences proptypes / add lowercase

### PR DESCRIPTION
**Pull request by igoroctaviano**
Improve hotkeyPreferences proptypes and lowercase event keys to guard
against different browsers casings

### CODE REVIEW

File: platform/ui/src/components/userPreferencesForm/HotKeysPreferences.js [hunk 1]
```diff
@@ -10,7 +10,13 @@ import PropTypes from 'prop-types';
...
 export class HotKeysPreferences extends Component {
   static propTypes = {
-    hotkeyDefinitions: PropTypes.array.isRequired,
+    hotkeyDefinitions: PropTypes.arrayOf(
+      PropTypes.shape({
+        commandName: PropTypes.string,
+        keys: PropTypes.arrayOf(PropTypes.string),
+        label: PropTypes.string,
+      })
+    ).isRequired,
```
Comment (dannyrb) [inline platform/ui/src/components/userPreferencesForm/HotKeysPreferences.js]: Just curious, what happens if `hotkeyDefinitions` is an empty array?

File: platform/ui/src/components/userPreferencesForm/UserPreferences.js [hunk 2]
```diff
@@ -15,7 +15,13 @@ export class UserPreferences extends Component {
...
   // TODO: Make this more generic. Tabs should not be restricted to these entries
   static propTypes = {
-    hotkeyDefinitions: PropTypes.array.isRequired,
+    hotkeyDefinitions: PropTypes.arrayOf(
+      PropTypes.shape({
+        commandName: PropTypes.string,
+        keys: PropTypes.arrayOf(PropTypes.string),
+        label: PropTypes.string,
+      })
+    ),
```
Comment (dannyrb) [inline platform/ui/src/components/userPreferencesForm/UserPreferences.js]: Why required above, but not here?

File: platform/ui/src/components/userPreferencesForm/UserPreferencesForm.js [hunk 3]
```diff
@@ -15,7 +15,13 @@ class UserPreferencesForm extends Component {
...
     onResetToDefaults: PropTypes.func,
     windowLevelData: PropTypes.object,
-    hotkeyDefinitions: PropTypes.array,
+    hotkeyDefinitions: PropTypes.arrayOf(
+      PropTypes.shape({
+        commandName: PropTypes.string,
+        keys: PropTypes.arrayOf(PropTypes.string),
+        label: PropTypes.string,
+      })
+    ),
```
Comment (dannyrb) [inline platform/ui/src/components/userPreferencesForm/UserPreferencesForm.js]: Why required above, but not here?

File: platform/ui/src/components/userPreferencesForm/HotKeysPreferences.js (hunk 1 above)
Comment (igoroctaviano) [reply platform/ui/src/components/userPreferencesForm/HotKeysPreferences.js]: The component will be rendered without any hotkey field. Do you think it's worth to conditionally render an informative text that no hotkey defs were found?

Comment (dannyrb) [reply platform/ui/src/components/userPreferencesForm/HotKeysPreferences.js]: I think something like:

> No hotkeys are configured for this application. Hotkeys can be configured in the application's `app-config.js` file.

Should be good enough for now. We can think of something better as a future enhancement.

File: platform/ui/src/components/userPreferencesForm/UserPreferencesForm.js (hunk 3 above)
Comment (igoroctaviano) [reply platform/ui/src/components/userPreferencesForm/UserPreferencesForm.js]: Done! thx.

File: platform/ui/src/components/userPreferencesForm/UserPreferences.js (hunk 2 above)
Comment (igoroctaviano) [reply platform/ui/src/components/userPreferencesForm/UserPreferences.js]: Done! thx.

File: platform/ui/src/components/userPreferencesForm/HotKeysPreferences.js (hunk 1 above)
Comment (igoroctaviano) [reply platform/ui/src/components/userPreferencesForm/HotKeysPreferences.js]: Im using the message directly for now, I have added the key/value to locize as well but I'm not sure about locals update.

Comment (dannyrb) [reply platform/ui/src/components/userPreferencesForm/HotKeysPreferences.js]: If it works locally, and it works in the deploy preview, then I think we're fine.

**Pull request merged**